planner/core/indexmerge_path.go

```diff
@@ -491,6 +491,75 @@ func (ds *DataSource) generateAndPruneIndexMergePath(indexMergeConds []expressio
 	return nil
 }
 
+// generateIndexMergeOnDNF4MVIndex generates IndexMerge paths for MVIndex upon DNF filters.
+/*
+	select * from t where ((1 member of (a) and b=1) or (2 member of (a) and b=2)) and (c > 10)
+		IndexMerge(OR)
+			IndexRangeScan(a, b, [1 1, 1 1])
+			IndexRangeScan(a, b, [2 2, 2 2])
+			Selection(c > 10)
+				TableRowIdScan(t)
+	Two limitations now:
+	1). all filters in the DNF have to be used as access-filters: ((1 member of (a)) or (2 member of (a)) or b > 10) cannot be used to access the MVIndex.
+	2). cannot support json_contains: (json_contains(a, '[1, 2]') or json_contains(a, '[3, 4]')) is not supported since a single IndexMerge cannot represent this SQL.
```

Comment:
what about `json_overlaps(a, '[1,2]') and json_overlaps(a, '[3,4]')`, seems it can not supported either.

Reply:
We can, but only one of them can be used to construct the IndexMerge, so the final plan is like this:
```
IndexMerge
    IndexScan(a, 1)
    IndexScan(a, 2)
    Selection(json_overlaps(a, [3, 4])
        TableRowIDScan
```

But this can be easily solved by merge these two filter into one `json_overlaps(a, '[1, 2, 3, 4]')`.

Reply:
I agree with the plan, but we can not merge them into one. after merge `[1]` will be true, it is not the same as the original

Reply:
Got it. You are right.